python/python/lance/util.py

```diff
@@ -40,3 +45,71 @@ def sanitize_ts(ts: ts_types) -> datetime:
     elif not isinstance(ts, datetime):
         raise TypeError(f"Unrecognized version timestamp {ts} of type {type(ts)}")
     return ts
+
+
+class KMeans:
+    def __init__(self, k: int, metric_type: str = "l2", max_iters: int = 50):
```

Comment:
nit: it would be nice to spell our support options in the types so users don't have to look through the docs to figure it out.
```suggestion
    def __init__(self, k: int, metric_type: Literal["l2", "cosine", "dot"] = "l2", max_iters: int = 50):
```

Reply:
Thanks. fixed.